### PR TITLE
Add `_sum` for `KernelSum{<:AbstractVector}` and associated tests

### LINKED ISSUE #497: `KernelSum` and `KernelProduct` with `Vector` of `Kernel` fails
Right now these composites will not work if the container is not a `Tuple`.
See

```julia
k = SqExponentialKernel()
kk = KernelSum([k, k])
kk(2, 3) # Errors
```

### PR DESCRIPTION
**Pull request by theogf**
**Summary**
Add the `_sum` interface for vectors, allowing to use `Vector` in `KernelSum`. This is to fix #497 

**Proposed changes**

Add a method for `_sum`  and add the same tests existing for `KerrnelSum{<:Tuple}` to `KernelSum{<:AbstractVector}`.

* ...

**What alternatives have you considered?**
:shrug: 

**Breaking changes**
none

### CODE REVIEW

File: test/kernels/kernelsum.jl [hunk 1]
```diff
@@ -2,21 +2,24 @@
...
-    @test length(k) == 2
-    @test repr(k) == (
+    @test length(k) == length(kvec) == 2
+    @test repr(k) ==
+        repr(kvec) ==
         "Sum of 2 kernels:\n" *
         "\tLinear Kernel (c = 0.0)\n" *
-        "\tSquared Exponential Kernel (metric = Euclidean(0.0))"
-    )
+        "\tSquared Exponential Kernel (metric = Distances.Euclidean(0.0))"
```
Comment (theogf) [inline test/kernels/kernelsum.jl:18]:
```suggestion
        "\tSquared Exponential Kernel (metric = Euclidean(0.0))"
```

File: src/kernels/kernelsum.jl [hunk 2]
```diff
@@ -45,6 +45,7 @@ Base.length(k::KernelSum) = length(k.kernels)
 
 _sum(f, ks::Tuple, args...) = f(first(ks), args...) + _sum(f, Base.tail(ks), args...)
 _sum(f, ks::Tuple{Tx}, args...) where {Tx} = f(only(ks), args...)
+_sum(f, ks::AbstractVector, args...) = sum(k -> f(k, args...), ks)
```
Comment (devmotion) [inline src/kernels/kernelsum.jl:48]: Codecov says the line is not covered by tests?

Comment (theogf) [reply src/kernels/kernelsum.jl:48]: I see it being covered? https://app.codecov.io/gh/JuliaGaussianProcesses/KernelFunctions.jl/pull/547?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=checks&utm_campaign=pr+comments&utm_term=JuliaGaussianProcesses#431d7fc89a2f6028dfd213ee63091cca-L49

Comment (devmotion) [reply src/kernels/kernelsum.jl:48]: Strange, the Github annotation was complaining about this line.

Comment (devmotion) [reply src/kernels/kernelsum.jl:48]: I can still see it here: https://github.com/JuliaGaussianProcesses/KernelFunctions.jl/pull/547/files

Comment (devmotion) [reply src/kernels/kernelsum.jl:48]: But I trust the codecov webpage more, maybe the annotation is just outdated.

**Review by devmotion — APPROVED** [review]

**Pull request merged**